File: Chapter11/javalist/src/main/java/com/example/javalist/MainActivity.java
package com.example.javalist;

import android.os.Bundle;
import android.view.View;
import android.widget.AbsListView;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;


public class MainActivity extends AppCompatActivity {

    TextView textView;
    ListView listView;

    // ListView에  표시될 데이터 배열
    ArrayList<String> dataList = getStringList();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.textView);
        listView = findViewById(R.id.listView);

        //ArrayAdapter에 ListView의 하나의 아이템을 구성할 View와 데이터를 연결
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this,
                R.layout.listview,R.id.textViewRow, dataList);

        listView.setChoiceMode(AbsListView.CHOICE_MODE_SINGLE);

        // ListView에 ArrayAdapter 연결
        listView.setAdapter(adapter);

        // ListView의 각 항목을 클릭했을때, 이벤트 처리
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                //textView.setText(dataArr[position]);
                textView.setText(dataList.get(position));
            }
        });


    }

    // ListView에 표시될 데이터 배열 생성 메소드
    private String[] getStringArray() {
        String[] Temp = new String[100];
        for(int i = 0; i < Temp.length; i++){
            Temp[i] = "Arr 아이템" + (i+1) ;
        }
        return Temp;
    }

    // ListView에 표시될 데이터 ArrList 생성 메소드
    private ArrayList<String> getStringList() {
        ArrayList<String> arrayList = new ArrayList<String>();
        for(int i = 0; i < 100; i++){
            arrayList.add("리스트 아이템" + (i + 1) );
        }
        return arrayList;
    }
}

File: Chapter11/ex01_listview/src/main/java/com/example/ex01_listview/MainActivity.kt
package com.example.ex01_listview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    //DATA Arr


    private fun getStringArray(): Array<String> {

        return Array(100){ i->i.toString()+"hi"}
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var dataArr = getStringArray()

        val adapter = ArrayAdapter<String>(this, android.R.layout.simple_list_item_1,dataArr)


        // lastView ArrayAdapter connect

        listView.adapter = adapter

        listView.setOnItemClickListener{
            parent, view, position, id ->
            textView.setText(dataArr[position])
        }


    }
}
